### PR TITLE
[KINETIS] mcu/nxp: adding K8xF support. hw/bsp: adding frdm-k82f BSP

### PR DESCRIPTION
**Pull request by mlaz**
This PR depends on https://github.com/apache/mynewt-core/pull/2466
It adds MCU definitions for kinetis K8xF chips, namely K82f.

### CODE REVIEW

**Comment by utzig** [comment]
This is not properly rebased, you can just drop the first two commits.

File: .rat-excludes [hunk 1]
```diff
@@ -291,6 +291,15 @@ fsl_clock.c
...
 startup_MK64F12.S
+MK82FN256xxx15_flash.ld
+boot-MK82FN256xxx15_flash.ld
+startup_MK82F25615.S
+MK82F25615.h
+MK82F25615_features.h
+system_MK82F25615.h
+system_MK82F25615.c
+fsl_clock.h
+
```
Comment (utzig) [inline .rat-excludes:302]: Remove the empty line.

File: hw/mcu/nxp/kinetis/MK64F12/pkg.yml [hunk 2]
```diff
@@ -23,6 +23,4 @@ pkg.author: "Apache Mynewt <dev@mynewt.apache.org>"
 pkg.homepage: "http://mynewt.apache.org/"
 
 pkg.deps:
-    - "@apache-mynewt-core/hw/hal"
-    - "@apache-mynewt-core/hw/cmsis-core"
```
Comment (utzig) [inline hw/mcu/nxp/kinetis/MK64F12/pkg.yml:27]: Unrelated to K8x, move into another commit with rationale.

File: hw/mcu/nxp/kinetis/MK64F12/syscfg.yml [hunk 3]
```diff
@@ -17,3 +17,4 @@
 #
 syscfg.vals:
     OS_TICKS_PER_SEC: 1000
+    MCU_FLASH_MIN_WRITE_SIZE: 2
```
Comment (utzig) [inline hw/mcu/nxp/kinetis/MK64F12/syscfg.yml:20]: Move K6x changes into another commit.

**Review by utzig — CHANGES_REQUESTED** [review]
Add `hw/mcu/nxp/kinetis/MK8xF/MK82F` to `.style_ignored_dirs`.

File: hw/mcu/nxp/kinetis/MK8xF/include/mcu/frdm-k8xf_hal.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    int8_t pin_sda;
+    uint32_t frequency;
+};
+
+struct nxp_hal_spi_cfg {
+    uint32_t clk_pin;
+    uint32_t pcs_pin;
+    uint32_t sout_pin;
+    uint32_t sin_pin;
+};
```
[the remark this answers is not in the document]
Comment (mlaz) [reply hw/mcu/nxp/kinetis/MK8xF/include/mcu/frdm-k8xf_hal.h]: Done.

File: .rat-excludes (hunk 1 above)
Comment (mlaz) [reply .rat-excludes:302]: Done.

File: hw/mcu/nxp/kinetis/MK64F12/pkg.yml (hunk 2 above)
Comment (mlaz) [reply hw/mcu/nxp/kinetis/MK64F12/pkg.yml:27]: Done.

File: hw/mcu/nxp/kinetis/MK64F12/syscfg.yml (hunk 3 above)
Comment (mlaz) [reply hw/mcu/nxp/kinetis/MK64F12/syscfg.yml]: Done.

File: hw/bsp/frdm-k82f/src/hal_bsp.c [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+    rc = hal_spi_init(1, NULL, HAL_SPI_TYPE_MASTER);
+    assert(rc == 0);
+#endif
+#if MYNEWT_VAL(SPI_0_SLAVE)
+    rc = hal_spi_init(0, NULL, HAL_SPI_TYPE_SLAVE);
+    assert(rc == 0);
+#endif
+
+#if MYNEWT_VAL(SPI_1_SLAVE)
+    rc = hal_spi_init(0, NULL, HAL_SPI_TYPE_SLAVE);
```
Comment (utzig) [inline hw/bsp/frdm-k82f/src/hal_bsp.c:282]: `1`?

Comment (utzig) [reply hw/bsp/frdm-k82f/src/hal_bsp.c:282]: Shouldn't you also add initializations for SPI_2 master/slave?

Comment (mlaz) [reply hw/bsp/frdm-k82f/src/hal_bsp.c:282]: Done

Comment (utzig) [reply hw/bsp/frdm-k82f/src/hal_bsp.c:282]: I don't think you pushed your changes.

Comment (mlaz) [reply hw/bsp/frdm-k82f/src/hal_bsp.c]: oh, i failed to add this file to the commit. done now

**Review by utzig — APPROVED** [review]
LGTM
